File: dataset/human/java/file_1223.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    
    int N = sc.nextInt();
    int[] a = new int[N];

    long sum = 0;

    for (int i = 0; i < N; i++) {
      a[i] = sc.nextInt();
      sum += a[i];
    }

    sc.close();
    
    long min = Math.abs(sum - (2 * a[0]));
    long x = a[0];

    for (int i = 1; i < N-1; i++) {
      x += a[i];
      min = Math.min(min, Math.abs(sum - (2 * x)));
    }

    System.out.println(min);

  }

}

File: dataset/human/python/file_1223.py
import math
ans = math.inf
n = int(input())
c = list(map(int, input().split()))
acm = [0]*(n+1)
for i, v in enumerate(c, 1):
    acm[i] += acm[i-1] + v

s, a = 0, acm[n]
for i in range(1, n):
    s = acm[i]
    ans = min(ans, abs((a-s)-s))

print(ans)
